include/ebpf_structs.h

```diff
@@ -33,6 +33,9 @@ typedef enum bpf_map_type
     BPF_MAP_TYPE_RINGBUF = 13          ///< Ring buffer.
 } ebpf_map_type_t;
 
+#define BPF_MAP_TYPE_PER_CPU(X) \
+    ((X) == BPF_MAP_TYPE_PERCPU_HASH || (X) == BPF_MAP_TYPE_PERCPU_ARRAY || (X) == BPF_MAP_TYPE_LRU_PERCPU_HASH)
```

Comment:
Is there somewhere in [libs/execution_context/ebpf_maps.c](https://github.com/microsoft/ebpf-for-windows/pull/1423/files#diff-ae8a7da4edcb9e1f7c73d5704cb00d475c4f847515c8d97c08a178a326ccb27c) where we can assert that the per_cpu field is set to match this macro, so if someone adds one there and fails to update this, the assert will fail?

Reply:
Added assert at map creation time. Ideally, this should be a static_assert, but this is C. C_ASSERT requires the properties to be "const" by C rules, which are more restrictive than C++ constexpr rules.